### DISCUSSION TITLE
Entry denied due to maximum capacity

### DISCUSSION THREAD

**Discussion by luzmagurzua** [IIC2133-PUC/2024-2#30] [Tarea 0] [answered]
Hola, me pasa que el programa se cae al segundo intento de entrar cuando la sala está llena, por ejemplo si el output esperado (correcto) es: 
1.Penguin 19 has entered with 2 puffles
2.Penguin 11 has entered with 10 puffles
3.Entry denied due to maximum capacity
4.Entry denied due to maximum capacity
5.Penguin 12 bought 161
6.72 ran away from 2

mi programa imprime el primer "entry denied..." y se para. Este seria mi output actual:
1.Penguin 19 has entered with 2 puffles
2.Penguin 11 has entered with 10 puffles
3.Entry denied due to maximum capacity
y no imprime nada más 

Con otros tests que no terminan en ese output si me funciona, entonces, probablemente sea un error del manejo del flujo? porque lo he modificado hartas veces, pero sigue sin funcionar :(

**Comment by waknaudt** [comment]
Quiza no estes leyendo los puffles de los pinguinos cuya entrada ha sido denegada y despues el programa interprete esos puffles como un evento.

**Comment by CataAlamos** [comment, marked as answer]
Hola! Intenta correrlo con Valgrind, algunos errores de memoria detienen la ejecución del programa.
